modules/host/contractcreation.go

```diff
@@ -66,10 +66,10 @@ func (h *Host) considerTerms(terms modules.ContractTerms) error {
 	case terms.MissedProofAddress != consensus.ZeroAddress:
 		return errors.New("burn payout needs to go to the zero address")
 
-	case terms.Price < h.Price:
+	case terms.Price.Cmp(h.Price) < 0:
 		return errors.New("price does not match host settings")
 
-	case terms.Collateral > h.Collateral:
+	case terms.Collateral.Cmp(h.Collateral) > 0:
```

Comment:
I didn't appreciate the value of `Cmp` and `Sign` until I had to convert all of these. The return value of `Cmp` and `Sign` makes it easy because you can just move the comparison operator to the right and compare to 0.